### DISCUSSION TITLE
Improve feedback for missing permissions on GitLab

### DISCUSSION THREAD

**Discussion by krichter722** [renovatebot/renovate#28848] [Suggest an Idea]
### Tell us more.

When `renovate` is missing permissions for a repository the output in the logs is similar to

```
DEBUG: Autodiscovering GitLab repositories
DEBUG: hostRules: applying Bearer authentication for gitlab.com
DEBUG: Discovered 6 project(s)
DEBUG: Autodiscovered 6 repositories
DEBUG: Applying autodiscoverFilter
       "autodiscoverFilter": "[...]"
DEBUG: None of the discovered repositories matched the filter
```

when all repos are unter `myorg/`. Users usually start with a complex filter to try renovate in a subgroup. I then hit this error and started to trial and error regex modifications until I finally narrowed it down to `myorg/*` and finally omitting `autodiscoveryFilter`.

Through a stroke of luck I figured out that the issue has nothing to do with the repositories not being found, but that it must have been the GitLab Group Access token sitting too low on the group hierarchy, thus not having permissions to read the repos.

After moving it up, I could discover all repositories and use the filter as intended, but the 4 hours until then are lost. Very helpful would have been: `6 of the discovered 6 repositories matched the filter, but the GitLab Access Token is lacking permission for them`.

**Comment by viceice** [comment]
https://github.com/renovatebot/renovate/blob/ea4469a298d0bcbd902766f7f2a40cb1e3c45b11/lib/workers/global/autodiscover.ts#L47

if you see this message, then the platform didn't return any repo. filter is applied later

https://github.com/renovatebot/renovate/blob/ea4469a298d0bcbd902766f7f2a40cb1e3c45b11/lib/workers/global/autodiscover.ts#L62


you should run with debug log level when starting on renovate or changing settings.
